chainerx_cc/chainerx/routines/linalg.cc

```diff
@@ -325,4 +326,63 @@ Array PseudoInverse(const Array& a, float rcond) {
     return out;
 }
 
+std::tuple<Array, Array> Eigh(const Array& a, const std::string& uplo) {
+    CheckRankTwoArray(a);
+    CheckSquareMatrix(a);
+
+    Array w = Empty(Shape{a.shape()[0]}, a.dtype(), a.device());
+    Array v = Empty(a.shape(), a.dtype(), a.device());
+
+    {
+        NoBackpropModeScope scope{};
+        a.device().backend().CallKernel<SyevdKernel>(a, w, v, uplo, /*compute_eigen_vector=*/true);
+        v = v.Transpose();
+    }
+
+    // Reference:
+    // Section 3.1 https://people.maths.ox.ac.uk/gilesm/files/NA-08-01.pdf
+    {
+        BackwardBuilder bb{"eigh", a, {w, v}};
+        if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+            bt.Define([a_tok = bb.RetainInput(0), w_tok = bb.RetainOutput(0), v_tok = bb.RetainOutput(1)](BackwardContext& bctx) {
+                const Array& a = bctx.GetRetainedInput(a_tok);
+                const Array& w = bctx.GetRetainedOutput(w_tok);
+                const Array& v = bctx.GetRetainedOutput(v_tok);
+
+                const Array& gw = bctx.output_grad(0).has_value() ? *bctx.output_grad(0) : Zeros(w.shape(), a.dtype(), a.device());
+                const Array& gv = bctx.output_grad(1).has_value() ? *bctx.output_grad(1) : Zeros(v.shape(), a.dtype(), a.device());
+
+                Array vt = v.Transpose();
+
+                Array f = ExpandDims(w, 0) - ExpandDims(w, 1);
+                // Invert values of F, and fill the diagonal with 0s.
+                // F has 0s on the diagonal, therefore fill it first with infinity.
```

Comment:
Let's reduce confusion by using the same "f" instead of "F". If you want to clarify that it's referring to `f`, you could surround it with ticks, as in standard markdown.